pam/internal/adapter/newpasswordmodel.go

```diff
@@ -72,26 +73,53 @@ func (m newPasswordModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 
 	case tea.KeyMsg: // Key presses
 		switch msg.String() {
-		case "enter":
-			if m.focusIndex >= len(m.focusableModels) {
+		case "tab":
+			// Only allow tabbing if the form is skippable
+			if !m.skippable {
 				return m, nil
 			}
+
+			m.errorMsg = ""
+
+			// Only allow tabbing if no password was entered
+			for _, pe := range m.passwordEntries {
+				if pe.Value() != "" {
+					return m, nil
+				}
+			}
+
+			if m.focusIndex == 0 {
+				return m, m.focusPrevField()
+			}
+			return m, m.focusNextField()
+
+		case "enter":
 			entry := m.focusableModels[m.focusIndex]
 			switch entry := entry.(type) {
 			case *textinputModel:
-				// Check if the password is empty
-				if m.passwordEntries[0].Value() == "" {
-					m.Clear()
-					return m, sendEvent(errMsgToDisplay{msg: "Password must not be empty"})
+				m.errorMsg = ""
+
+				// First entry is focused
+				if m.focusIndex == 0 {
+					// Check if the password is empty
+					if m.passwordEntries[0].Value() == "" {
+						m.Clear()
+						return m, sendEvent(errMsgToDisplay{msg: "Password must not be empty"})
+					}
```

Comment:
FYI this is also done by pwqualitycheck, so not sure we need to replicate it here.